e2etest/bgptests/bgp.go

```diff
@@ -147,6 +164,134 @@ var _ = ginkgo.Describe("BGP", func() {
 			}),
 	)
 
+	ginkgo.Context("GracefulRestart support", func() {
+		ginkgo.BeforeEach(func() {
+			for _, c := range FRRContainers {
+				c.NeighborConfig.GracefulRestart = true
+			}
+		})
+
+		ginkgo.AfterEach(func() {
+			for _, c := range FRRContainers {
+				c.NeighborConfig.GracefulRestart = false
+			}
+		})
+
+		ginkgo.DescribeTable("dataplane should keep working when speaker restarts", func(pairingIPFamily ipfamily.Family, poolAddresses []string, tweak testservice.Tweak) {
+			_, svc := setupBGPService(cs, testNamespace, pairingIPFamily, poolAddresses,
+				FRRContainers, func(svc *corev1.Service) {
+					testservice.TrafficPolicyCluster(svc)
+					tweak(svc)
+				})
+			defer testservice.Delete(cs, svc)
+
+			allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			testservice.ValidateDesiredLB(svc)
+
+			for _, c := range FRRContainers {
+				validateService(svc, allNodes.Items, c)
+			}
+
+			// checkSVC := func(timeout time.Duration) error {
+			//	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+			//	defer cancel()
+			//	eg := new(errgroup.Group)
+			//	for _, c := range FRRContainers {
+			//		c := c // https://go.dev/doc/faq#closures_and_goroutines
+			//		eg.Go(func() error {
+			//			return validateServiceNoWait(svc, allNodes.Items, c)
+			//		})
+			//	}
+			//
+			//	done := make(chan error, 1)
+			//	go func() {
+			//		done <- eg.Wait()
+			//	}()
+			//
+			//	select {
+			//	case err := <-done:
+			//		return err
+			//	case <-ctx.Done():
+			//		return fmt.Errorf("checkSVC took more than %v second to return", timeout)
+			//	}
+			// }
+			//
+			//var monitoringError error
+			//
+			// ctx, cancel := context.WithCancel(context.Background())
+			// defer cancel()
+			// // Start a goroutine with a monitoring loop that:
+			// // - calls checkSVC() aggressively to catch fw-state bit regression
+			// // - if checkSVC() fails outside ErrStaleRoute, it appends to monitoringError
+			// // - gracefully exists on ctx cancel
+			// go func() {
+			//	//  http://onsi.github.io/ginkgo/#mental-model-how-ginkgo-handles-failure
+			//	defer ginkgo.GinkgoRecover()
+			//	ticker := time.NewTicker(time.Second)
+			//	defer ticker.Stop()
+			//	for {
+			//		select {
+			//		case <-ctx.Done():
+			//			return
+			//		case <-ticker.C:
+			//			fmt.Println("ticker in")
+			//			if err := checkSVC(2 * time.Second); err != nil {
+			//				if !errors.Is(err, ErrStaleRoute) {
+			//					monitoringError = errors.Join(monitoringError, err)
+			//				}
+			//			}
+			//		}
+			//	}
+			// }()
+
+			Expect(metallb.RestartSpeakerPodsNoWait(cs)).NotTo(HaveOccurred())
+
+			Eventually(func() error {
+				c := func() error {
```

Comment:
why do you need to assign to a variable? Can't you just iterate over the containers and assert from there?

Reply:
I find easier to read but not strong opinion, I will change it

Reply:
done